core/templates/dev/head/pages/topic-editor-page/modal-templates/select-skill-modal.template.html

```diff
@@ -1,31 +1,69 @@
-  <div ng-if="skillSummaries.length == 0">
-    You can only create questions for topics that have atleast one skill
-    attached to it. Please link a skill to this topic before adding questions.
+<!-- Select Skill View -->
+<div ng-if="!selectDifficulty">
+  <div class="modal-header">
+    <h3 ng-if="skillSummaries.length == 0">
+      No skills linked to topic
+    </h3>
+    <h3 ng-if="skillSummaries.length > 0">
+      Select the skill to link the question to:
+    </h3>
+  </div>
+  <div class="modal-body select-skill-modal">
+    <div ng-if="skillSummaries.length == 0">
+      You can only create questions for topics that have atleast one skill
+      attached to it. Please link a skill to this topic before adding questions.
+    </div>
+    <div class="list-group">
+      <a class="list-group-item protractor-test-skill-item"
+         ng-repeat="skillSummary in skillSummaries track by $index"
+         ng-click="selectOrDeselectSkill(skillSummary.getId(), $index)"
+         ng-class="{'list-group-item-success': skillSummary.isSelected}">
+        <label class="list-group-item-heading" style="font-size: 1em;"><[skillSummary.getDescription()]></label>
+      </a>
+    </div>
   </div>
-  <div class="list-group">
-    <a class="list-group-item protractor-test-skill-item"
-       ng-repeat="skillSummary in skillSummaries track by $index"
-       ng-click="selectOrDeselectSkill(skillSummary.getId(), $index)"
-       ng-class="{'list-group-item-success': skillSummary.isSelected}">
-      <label class="list-group-item-heading" style="font-size: 1em;"><[skillSummary.getDescription()]></label>
-    </a>
+  <div class="modal-footer" ng-switch="skillSummaries.length > 0">
+    <div ng-switch-when="true">
+      <button class="btn btn-default" ng-click="cancel()"> Cancel </button>
+      <button class="btn btn-success protractor-test-confirm-skill-button" ng-disabled="selectedSkillId === null" ng-click="next()"> Next </button>
+    </div>
+    <div ng-switch-when="false">
+      <button class="btn btn-default" ng-click="ok()"> Ok </button>
+    </div>
   </div>
 </div>
-<div class="modal-footer" ng-switch="skillSummaries.length > 0">
-  <div ng-switch-when="true">
-    <button class="btn btn-default" ng-click="cancel()"> Cancel </button>
-    <button class="btn btn-success protractor-test-confirm-skill-button" ng-disabled="selectedSkillId === null" ng-click="done()"> Done </button>
+<!-- Select Difficulty View -->
+<div ng-if="selectDifficulty">
+  <div class="modal-header" ng-if="selectDifficulty">
+    <h3 ng-if="selectedSkillIds.length == 0">
+      Select skills you would like to link to the question.
+    </h3>
+    <h3 ng-if="selectedSkillIds.length > 0">
+      Select the difficulty per skill:
+    </h3>
+  </div>
+  <div class="modal-body select-skill-modal">
+    <div ng-if="selectedSkillIds.length == 0">
+      You can only set difficulty after selecting skills linked to the question. Please go back and link skill(s) to this question.
```

Comment:
Can we have the button disabled if user didn't select any skill instead of having this warning afterwards?

Reply:
So that's already being done (see ng-disabled in the 'Next' button, this is just a failsafe as well.

Reply:
For some reason ng-disabled isn't working?

Reply:
Done

Reply:
Do we still need this warning if the button is disabled in this case?

Reply:
I want to still keep it in case there are bugs like this in the future (where the button wasn't getting disabled). Also in case someone is able to force a button click, we don't want the next modal to break.

Reply:
Ok that makes sense!